src/drivers/queue_interop/Queue.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace yii\queue\queue_interop;
+
+use Interop\Amqp\AmqpContext;
+use Interop\Amqp\AmqpQueue;
+use Interop\Queue\PsrConnectionFactory;
+use Interop\Queue\PsrContext;
+use Interop\Queue\PsrProducer;
+use yii\base\NotSupportedException;
+use yii\queue\cli\Queue as CliQueue;
+
+class Queue extends CliQueue
+{
+    /**
+     * {@inheritdoc}
+     */
+    public $commandClass = Command::class;
+
+    /**
+     * @var string
+     */
+    public $factoryClass = null;
+
+    /**
+     * @var array
+     */
+    public $factoryConfig = [];
+
+    /**
+     * @var string
+     */
+    public $queueName = 'queue';
+
+    /**
+     * @var PsrContext
+     */
+    private $context;
+
+    /**
+     * @var PsrProducer
+     */
+    private $producer;
+
+    /**
+     * Listens queue and runs new jobs.
+     */
+    public function listen()
+    {
+        $consumer = $this->getContext()->createConsumer(
+            $this->getContext()->createQueue($this->queueName)
+        );
+
+        while (true) {
+            if ($message = $consumer->receive()) {
+                list($ttr, $body) = explode(';', $message->getBody(), 2);
+                if ($this->handleMessage(null, $body, $ttr, 1)) {
+                    $consumer->acknowledge($message);
+                } else {
+                    $consumer->reject($message, true);
+                }
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function pushMessage($message, $ttr, $delay, $priority)
+    {
+        if ($delay) {
+            throw new NotSupportedException('Delayed work is not supported in the driver.');
```

Comment:
@ASKozienko Could you please implement this (since their support has been merged https://github.com/php-enqueue/enqueue-dev/pull/149)